nbbuild/antsrc/org/netbeans/nbbuild/CheckLinks.java

```diff
@@ -633,8 +645,33 @@ final Boolean isOk (URI u) throws BuildException {
                 throw new BuildException ("Each filter must have pattern attribute");
             }
             
-            if (pattern.matcher (u.toString ()).matches ()) {
-                log ("Matched " + u + " accepted: " + accept, org.apache.tools.ant.Project.MSG_VERBOSE);
+            if (pattern.matcher(u.toString()).matches()) {
+                log("Matched " + u + " accepted: " + accept, org.apache.tools.ant.Project.MSG_VERBOSE);
+                if (externallinksdump != null) {
+                    try {
+                        String dummyName;
+                        if (accept) {
+                            dummyName = "acceptednetbeans.txt";
+                        } else {
+                            dummyName = "rejectednetbeans.txt";
+                        }
+                        Path dumppath = externallinksdump.toPath().resolve(dummyName);
+                        if (Files.notExists(dumppath)) {
+                            Files.createDirectories(dumppath.getParent());
+                            Files.createFile(dumppath);
+                        }
+                        List<String> existingEntries = Files.readAllLines(externallinksdump.toPath().resolve(dummyName));
```

Comment:
existing entries variable seems unneeded.
would
Set<String> sortedEntries = new TreeSet<>(Files.readAllLines(externallinksdump.toPath().resolve(dummyName)));
rork?

Reply:
will change that thanks